src/harness/sas_test_harness.py

```diff
@@ -0,0 +1,364 @@
+  """SasTestHarnessServerHandler class is inherited with BaseHTTPRequestHandler
+  to serve HTTP Response.
+  """
+
+  def __checkDumpFileExist(self, url_id):
+    """Check the url_id received in GET request passed as argument to check corresponding
+    dump file exist in that path.
+    """
+    return os.path.exists(os.path.join(self.server.base_path, '{}.json'.format(url_id)))
+
+  def __getDumpFilePath(self, url_id):
+    """ Return the absolute path of FAD records that are going
+    to be generated.
+    """
+    return os.path.join(self.server.base_path, '{}.json'.format(url_id))
+
+  def do_GET(self):
+    """Handles Pull/GET Request and returns Path of the Request to callback Method."""
+
+    if "dump" in self.path:
+      self.send_response(200)
+      self.send_header('Content-type', 'application/json')
+      self.end_headers()
+      self.wfile.write(json.load(open(self.__getDumpFilePath('FAD'))))
+    else:
+      url_encoded_id = self.path.split('/')[-1]
+      if not self.__checkDumpFileExist(url_encoded_id):
+        self.send_error(404)
+        return
+      self.send_response(200)
+      self.send_header('Content-type', 'application/json')
+      self.end_headers()
+      self.wfile.write(json.load(open(self.__getDumpFilePath(url_encoded_id))))
+
+# Helper functions for FAD related tests.
+def generateCbsdReferenceId(fcc_id, serial_number):
+  """creates encoded cbsd_reference_id using sha1 with fcc_id and serial number of cbsd.
+  """
+  return str(fcc_id + '/' + str(hashlib.sha1(serial_number).hexdigest())).encode('utf-8')
+
+def generateCbsdRecords(cbsd_records, grant_records_list):
+  """ Generates the cbsdData Object by combining cbsdrecord and grantrecord based on SAS-SAS TS.
+
+  Args:
+     cbsd_records: list of cbsdData objects that includes one or more grants object.
+     grant_records_list: list of muliple grant records for all cbsd devices.
+       For an example : muliple grants [ [grant1, grant2], [grant1, grant2] ] are
+       associated with each cbsd records [ cbsd1, cbsd2 ].
+
+  Returns:
+     It returns list of CBSDData Object based on SAS-SAS TS.
+  """
+
+  cbsd_records_list = []
+  for index, (cbsd_record, grant_records) in enumerate(zip(cbsd_records, grant_records_list)):
+    encoded_cbsd_id = generateCbsdReferenceId(cbsd_record['fccId'],
+                                         cbsd_record['cbsdSerialNumber'])
+
+    # Calculate grant_expiry time for grant objects if grant expiry time may
+    # be incorrect or missing.
+    grant_time_stamp = datetime.now().today() + timedelta(hours=24)
+    grant_expiry_time = time.strftime(grant_time_stamp.replace(microsecond=0).isoformat()) + 'Z'
+
+    internal_data = {
+        'id': 'cbsd/{}'.format(encoded_cbsd_id),
+        'registration': {
+            'fccId': cbsd_record['fccId'],
+            'cbsdCategory': cbsd_record['cbsdCategory'],
+            'callSign': cbsd_record['callSign'],
+            'userId': cbsd_record['userId'],
```

Comment:
User ID may not be exchanged via SAS-SAS so this should be removed.

Reply:
Removed   the userID